cql3/restrictions/statement_restrictions.cc

```diff
@@ -1495,6 +1490,42 @@ void statement_restrictions::prepare_indexed(const schema& idx_tbl_schema, bool
         const auto pos = _schema->position(*col) + 1;
         (*_idx_tbl_ck_prefix)[pos] = replace_column_def(e, &idx_tbl_schema.clustering_column_at(pos));
     }
+
+    add_clustering_restrictions_to_idx_ck_prefix(idx_tbl_schema);
+
+    (*_idx_tbl_ck_prefix)[0] = binary_operator(
+            column_value(token_column),
+            oper_t::EQ,
+            // TODO: This should be a unique marker whose value we set at execution time.  There is currently no
+            // handy mechanism for doing that in query_options.
+            ::make_shared<constants::value>(raw_value::make_unset_value()));
+}
+
+void statement_restrictions::prepare_indexed_local(const schema& idx_tbl_schema) {
+    if (!_partition_range_is_simple) {
+        return;
+    }
+
+    // Local index clustering key is (indexed column, base clustering key)
+    _idx_tbl_ck_prefix = std::vector<expr::expression>();
```

Comment:
Please also change the blurb on _idx_tbl_ck_prefix in statement_restrictions.hh; it currently describes only the global-index case.